src/components/disclosure/components/disclosure-activator/types.ts

```diff
@@ -1,3 +1,7 @@
+import { ForwardedRef } from 'react'
+
+type DisclosureActivatorForwardedRef = ForwardedRef<HTMLButtonElement>
```

Comment:
A few reasons!

- In `NavigationDisclosureActivator` [where this new type is referenced](https://github.com/hashicorp/dev-portal/blob/e43398105a92a2b84b0a1dba13c7b2f0e0016af9/src/components/navigation-disclosure/components/navigation-disclosure-activator/index.tsx#L22),
  - Having the type be explicit when you're reading the code means you don't have to hover over anything to see what the inferred type is (and you don't have to try to remember it after hovering either -- think of it like throwing away some food packaging but needing to check the box a few times because you forgot the instructions)
  - We could just put `ForwardedRef<HTMLButtonElement>`, but that means if we change the type of the ref on `DisclosureActivator` we have to update the type in two places.
- The shared type also helps express at a glance what `NavigationDisclosureActivator` is going to do with the `ref`. Without looking too deeply at the code, inferences can be made that the `ref` is going to be passed to a `DisclosureActivator` component.

Reply:
Yep I saw where it was referenced! I think my main gripe is that the type alias introduces an additional layer of indirection when I see it, as I don't know what type the ref is, and hovering over places where it is inferred as this type might only show `DisclosureActivatorForwardedRef` instead of the underlying `ForwardedRef<HTMLButtonElement>`.

Thanks for the insight though! Fine by me if you feel it's the best way to type this 👍